identity_credential/src/validator/errors.rs

```diff
@@ -139,4 +140,65 @@ impl Display for CompoundPresentationValidationError {
   }
 }
 
-impl std::error::Error for CompoundPresentationValidationError {}
+impl Error for CompoundPresentationValidationError {}
+
+#[derive(Debug, thiserror::Error)]
+/// An error caused by a failure to verify a Domain Linkage configuration or credential.
+pub struct DomainLinkageVerificationError {
```

Comment:
Should we mark this `[non_exhaustive]`? Probably not, but just want to make sure. @olivereanderson any opinions?

Reply:
Maybe it is best to instead make the fields private and introduce a getter for `cause` (the `source` is retrievable through the error trait).